File: Farhaan/Week-2 Problems/container-with-most-water.cpp
class Solution {
public:
int maxArea(vector<int>& height) {
	int n=height.size();
	int maxx=0,left = 0,right=n-1;
	while(left<right){
		
		int area = abs(left-right)*min(height[left],height[right]);
		
		if(height[left]<=height[right]){
			left++;
		}else
			right--;
			//cout<<left<<" "<<right<<" "<<area<<endl;
		maxx = max(area,maxx);
	}
	return maxx;

	}
};

File: Ashish/Week 2/containerwithmostwater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i,j = 0,len(height) - 1
        max1 = 0
        while i<j:
            h = min(height[i],height[j])
            l = j - i
            if(max1 <= h*l):
                max1 = h*l
            if(height[i] < height[j]):
                i+=1
            else:
                j-=1
        return max1